pilot/pkg/bootstrap/server.go

```diff
@@ -325,6 +334,40 @@ func GetMeshConfig(kube kubernetes.Interface, namespace, name string) (*v1.Confi
 	return config, mesh, nil
 }
 
+// GetMeshNetworks fetches the MeshNetworks configuration from Kubernetes ConfigMap.
+func GetMeshNetworks(kube kubernetes.Interface, namespace, name string) (*v1.ConfigMap, *meshconfig.MeshNetworks, error) {
+
+	if kube == nil {
+		defaultMeshNetworks := model.EmptyMeshNetworks()
+		return nil, &defaultMeshNetworks, nil
+	}
+
+	config, err := kube.CoreV1().ConfigMaps(namespace).Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			defaultMeshNetworks := model.EmptyMeshNetworks()
+			return nil, &defaultMeshNetworks, nil
+		}
+		return nil, nil, err
+	}
+
+	// values in the data are strings, while proto might use a different data type.
+	// therefore, we have to get a value by a key
+	cfgYaml, exists := config.Data[MeshNetworksConfigMapKey]
+	if !exists {
+		return nil, nil, fmt.Errorf("missing configuration map key %q", MeshNetworksConfigMapKey)
+	}
+
+	mesh, err := model.LoadMeshNetworksConfig(cfgYaml)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO Add watcher to track changes in the config map file and re-load it
```

Comment:
Keep in mind that without this todo, you can't add/remove clusters dynamically. So, its a pretty big usability limitation

Reply:
I will add it. I think it's a simple `fsnotify` on the file?

Reply:
Use https://github.com/istio/istio/tree/master/pilot/pkg/config/monitor

Reply:
Trying to understand how to use the Monitor for the watch but it feels like killing a fly with a sledgehammer 🙂